### PR TITLE
AWS - Emr security-configuration resource

### PR DESCRIPTION
**Pull request by Lucas-Irvine**
Create a new EMR security configuration resource to filter for EMR security configurations, for example to check ones that do not have encryption fields enabled.

### CODE REVIEW

File: c7n/resources/emr.py [hunk 1]
```diff
@@ -284,3 +285,46 @@ def query(self):
...
+
+    def augment(self, resources):
+
+        def _augment(r):
+            # Convert the json string to dict
+            r['SecurityConfiguration'] = json.loads(r['SecurityConfiguration'])
+            return r
+
+        # Describe notebook-instance & then list tags
+        resources = super(EMRSecurityConfiguration, self).augment(resources)
```
Comment (JohnHillegass) [inline c7n/resources/emr.py:314]: This can be done with super().augment(resources) as is the preferred method now since we killed py2.x compatibility

Comment (Lucas-Irvine) [reply c7n/resources/emr.py:314]: Thanks, I didn't know that! Will change

File: c7n/resources/emr.py [hunk 2]
```diff
@@ -284,3 +285,46 @@ def query(self):
...
+    def augment(self, resources):
+
+        def _augment(r):
+            # Convert the json string to dict
+            r['SecurityConfiguration'] = json.loads(r['SecurityConfiguration'])
+            return r
+
+        # Describe notebook-instance & then list tags
+        resources = super().augment(resources)
+        return list(map(_augment, resources))
```
Comment (kapilt) [inline c7n/resources/emr.py:315]: this seems like it would simpler/more readable as a for loop for deserializing the json

Comment (Lucas-Irvine) [reply c7n/resources/emr.py]: Totally Agree, made the change

**Pull request merged**